instrumentation/mongo/lib/opentelemetry/instrumentation/mongo/middlewares/subscriber.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require_relative 'command_serializer'
+
+module OpenTelemetry
+  module Instrumentation
+    module Mongo
+      module Middlewares
+        # Event handler class for Mongo Ruby driver
+        class Subscriber
+          THREAD_KEY = :__opentelemetry_mongo_spans__
+
+          def started(event) # rubocop:disable Metrics/AbcSize
+            # start a trace and store it in the current thread; using the `operation_id`
+            # is safe since it's a unique id used to link events together. Also only one
+            # thread is involved in this execution so thread-local storage should be safe. Reference:
+            # https://github.com/mongodb/mongo-ruby-driver/blob/master/lib/mongo/monitoring.rb#L70
+            # https://github.com/mongodb/mongo-ruby-driver/blob/master/lib/mongo/monitoring/publishable.rb#L38-L56
+            # TODO: decide naming convention
+            name = 'mongo.cmd'
+            # name = "#{event.database_name}.#{event.command_name}"
+            span = tracer.start_span(name, kind: :client)
+            set_span(event, span)
+
+            serialized_command = CommandSerializer.new(event.command).serialize
+            collection = get_collection(event.command)
+            span.set_attribute('component', 'mongo-ruby')
+            span.set_attribute('db.type', 'mongodb')
+            span.set_attribute('db.instance', event.database_name)
+            span.set_attribute('mongo.command', event.command_name)
+            span.set_attribute('mongo.collection', collection) if collection
+            span.set_attribute('db.statement', serialized_command)
+            span.set_attribute('mongo.request_id', event.request_id)
+            span.set_attribute('mongo.op_id', event.operation_id)
+            span.set_attribute('peer.hostname', event.address.host)
+            span.set_attribute('peer.port', event.address.port)
+          end
+
+          def failed(event)
+            finish_event('failed', event) do |span|
+              if event.is_a?(::Mongo::Monitoring::Event::CommandFailed)
+                span.set_attribute('error', true)
+                span.set_attribute('error.kind', 'CommandFailed')
+                span.set_attribute('message', event.message)
+              end
+            end
+          end
+
+          def succeeded(event)
+            finish_event('succeeded', event) do |span|
+              n = event.reply.fetch('n', nil)
+              span.set_attribute('mongo.n', n) unless n.nil?
+            end
+          end
```

Comment:
```suggestion
          def succeeded(event)
          end
```
This isn't part of the OpenTelemetry semantic conventions for Mongo.

Reply:
Could we add this? This is the number of modified records. Again very useful to see in the trace and quickly explain latency.

Reply:
As above, could you please open an issue (or create a PR) for this in the spec repo? I think it'd require a prefix like `db.mongodb.` and should probably have a more meaningful name than `n`.

Reply:
Ditto to @fbogsany comments here. I think it's a reasonable thing to add but i think it would gain broader support using a more descripitve naming convention